### PR TITLE
Use enclave_info.toml instead of txt

### LINKED ISSUE #152: Use TOML format to store enclave info
Current enclave info is stored in the `enclave_info.txt` which is a customize format. I'm proposing to use TOML for enclave_info. Here is an example:

`enclave_info.toml`:

```toml
[acs]
mrsigner     = "83d719e77deaca1470f6baf62a4d774303c899db69020f9c70ee1dfc08c7ce9e"
enclave_hash = "045f15bc75fb3f021b8e96e91ffebc8137005836f8b1f497024896ba0f3e6cb3"

[fns]
mrsigner     = "83d719e77deaca1470f6baf62a4d774303c899db69020f9c70ee1dfc08c7ce9e"
enclave_hash = "188688c12b710ffa51ae0cf93ddca99f698cd198cbd4391ec44cf6d6091231ff"

[kms]
mrsigner     = "83d719e77deaca1470f6baf62a4d774303c899db69020f9c70ee1dfc08c7ce9e"
enclave_hash = "c208057b989f6c6b19da9ef014529246e8c7d1472f805307e74b209699f8ebf2"

[tdfs]
mrsigner     = "83d719e77deaca1470f6baf62a4d774303c899db69020f9c70ee1dfc08c7ce9e"
enclave_hash = "036e3910fed68acdd0a068e9b3e14b9f17ec04b2c666faeda7214bae645c25aa"

[tms]
mrsigner     = "83d719e77deaca1470f6baf62a4d774303c899db69020f9c70ee1dfc08c7ce9e"
enclave_hash = "69d9c9dd9ada2b0846d013a50a7b126721c9a07647f50d8a18ec5cb9361bc25a"
```

Deserialization test:

```rust
use serde_derive::{Deserialize};
use serde::{Deserialize, Deserializer};
use hex::FromHex;
use std::collections::HashMap;
use std::fs;

/// Deserializes a hex string to a `Vec<u8>`.
pub fn from_hex<'de, D>(deserializer: D) -> Result<Vec<u8>, D::Error>
where D: Deserializer<'de>
{
    use serde::de::Error;
    String::deserialize(deserializer)
        .and_then(|string| Vec::from_hex(&string).map_err(|err| Error::custom(err.to_string())))
}

#[derive(Debug, Deserialize)]
#[serde(transparent)]
struct EnclaveInfoToml(HashMap<String, EnclaveInfo>);

#[derive(Debug, Deserialize)]
struct EnclaveInfo {
    #[serde(deserialize_with = "from_hex")]
    mrsigner: Vec<u8>,
    #[serde(deserialize_with = "from_hex")]
    enclave_hash: Vec<u8>,
}

fn main() {
    let contents = fs::read_to_string("enclave_info.toml")
        .expect("Something went wrong reading the enclave info file.");
    let config: EnclaveInfoToml = toml::from_str(&contents).unwrap();
    println!("{:?}", config);
}
```

### PR DESCRIPTION
**Pull request by mssun**
## Description

This PR introduce TOML format enclave_info to replace txt.

Fix #152.

## Type of change (select applied and DELETE the others)

- [x] Breaking change: changes of enclave_info format

## How Has This Been Tested?
CI: https://ci.mesalock-linux.org/mssun/incubator-mesatee/126

## Checklist (check ALL before submitting PR, even not applicable)

- [x] Fork the repo and create your branch from `master`.
- [x] If you've added code that should be tested, add tests.
- [x] If you've changed APIs, update the documentation.
- [x] Ensure the tests pass (see CI results).
- [x] Make sure your code lints/format.

### CODE REVIEW

**Comment by mssun** [comment]
This PR also fix #156.

**Comment by mssun** [comment]
I'm done with this PR. Please review. The latest CI result can be seen here: https://ci.mesalock-linux.org/mssun/incubator-mesatee/129

I'll work on #158 once this PR is merged. Thanks.

**Comment by m4sterchain** [comment]
> I'm done with this PR. Please review. The latest CI result can be seen here: https://ci.mesalock-linux.org/mssun/incubator-mesatee/129
> 
> I'll work on #158 once this PR is merged. Thanks.

Great! I will review the PR later today.

File: mesatee_utils/teaclave_utils/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+struct EnclaveInfo {
+    #[serde(deserialize_with = "from_hex")]
+    mrsigner: SgxMeasure,
+    #[serde(deserialize_with = "from_hex")]
+    enclave_hash: SgxMeasure,
+}
+
+// This function fails when enclave info signatures mismatch hard-coded
+// auditor keys. We expect the program to crash in those cases
+pub fn load_and_verify_enclave_info(
```
[the remark this answers is not in the document]
Comment (mssun) [reply mesatee_utils/teaclave_utils/src/lib.rs]: Yes, this need to be rewritten later.

**Pull request merged**